src/System.Net.Http/src/System/Net/Http/SocketsHttpHandler/Http2Connection.cs

```diff
@@ -1330,23 +1318,33 @@ private void AbortStreams(Exception abortException)
             // we could hold pool lock while trying to grab connection lock in Dispose().
             _pool.InvalidateHttp2Connection(this);
 
+            List<Http2Stream> streamsToAbort = new List<Http2Stream>();
```

Comment:
How likely is it that there won't be any streams to abort?  Wondering if this should be lazily allocated inside the foreach below.

Reply:
In this case, not that likely; here we are dealing with IO failure, which probably means the server crashed.

In the GOAWAY case below, much more likely; this is probably because the server is gracefully closing an idle connection.

That said, the cost here doesn't seem significant.